explorer/nodewatch/nodewatch/RoutesFacade.py

```diff
@@ -94,17 +94,14 @@ def custom_filter(descriptor):
 		if order == 'random':
 			random.shuffle(nodes)
 
-		if limit is not None:
-			nodes = nodes[:limit]
-
-		return nodes
+		return nodes if limit == 0 else nodes[:limit]
 
 	def json_node(self, filter_field, public_key):
```

Comment:
you might want to consider checking if public_key is proper public key and returning 400 if not